### PR TITLE
PHPC-285: Windows compilation fixes for PHP 7

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-285

### CODE REVIEW

File: php_bson.h [hunk 1]
```diff
@@ -71,7 +71,7 @@ typedef struct {
 } php_phongo_bson_state;
 
 #if PHP_VERSION_ID >= 70000
-#define PHONGO_BSON_STATE_INITIALIZER  { {}, { PHONGO_TYPEMAP_NONE, NULL, PHONGO_TYPEMAP_NONE, NULL, PHONGO_TYPEMAP_NONE, NULL}, NULL}
+#define PHONGO_BSON_STATE_INITIALIZER  { {{ 0 }}, { PHONGO_TYPEMAP_NONE, NULL, PHONGO_TYPEMAP_NONE, NULL, PHONGO_TYPEMAP_NONE, NULL}, NULL}
```
Comment (jmikola) [inline php_bson.h]: @derickr: See http://stackoverflow.com/q/11551986 and http://stackoverflow.com/a/20405554. Windows required `{ 0 }` (as is already done in libmongoc), but the extra braces here are a concession for gcc's missing-braces warning. Windows seems to be OK with it, too, so I gathered this was preferable to disabling the warning altogether.

File: src/contrib/php-ssl.c [hunk 2]
```diff
@@ -30,6 +30,10 @@
 /* LCOV_EXCL_START */
 #include "php-ssl.h"
 
+#if defined(NETWARE) || (defined(PHP_WIN32) && defined(_MSC_VER) && _MSC_VER >= 1900)
+#define timezone _timezone  /* timezone is called _timezone in LibC */
+#endif
```
Comment (jmikola) [inline src/contrib/php-ssl.c]: See: http://lxr.php.net/xref/PHP_7_0/ext/openssl/openssl.c#62

File: php_bson.h (hunk 1 above)
Comment (jmikola) [inline php_bson.h]: Also, kudos to @ksuarz to the quick insight into this!

**Comment by derickr** [comment]
Lgtm

**Pull request merged**